File: twosum.js
let nums = [3,3]
let target = 6

const twoSum = (nums, target) => {
    for (let index1 = 0; index1 < nums.length; index1++) {
        const num1 = nums[index1];
        for (let index = 1; index < nums.length; index++) {
            const num2 = nums[index];
            if (target - num2 === num1) {
                console.log([index1, index])
            }       
        }  
    }
};

const sumTwo = function (nums, target) {
    var numObj = new Map();
    for (var i = 0; i < nums.length; i++) {
        numObj.set(nums[i], i);
    }

    for (var i = 0; i < nums.length; i++) {
        var complement = target - nums[i];
        if (numObj.has(complement) && numObj.get(complement) != i) {
            console.log([i, numObj.get(complement)]);
        }
    }
    console.log([]); 
};

// sumTwo(nums,target)

twoSum(nums, target)

File: twosum.py
def twoSum(nums, target):
    hashMap = {}  # Map to maintain nums O(1) look up

    for x in nums:  # Iterate through list

        y = target - x  # Complement

        if y in hashMap.keys():  # if Complement in keys return x, y
            return [x, y]
        else:
            hashMap[x] = True  # Else add to map and continue
    return []


testCase = {"array": [3, 5, -4, 8, 11, 1, -1, 6], "target": 10}


def run_tests():
    ans = twoSum(testCase['array'], testCase['target'])

    assert ans == [11, -1] or ans == [-1,
                                      11], f'Should be [11, -1], returned {ans}'

    print("All Tests Passed :)")


run_tests()
